DSharpPlus.Caching/Memory/MemoryCacheService.cs

```diff
@@ -76,16 +68,37 @@ public ValueTask CacheAsync<TItem>(AbstractCacheEntry entry)
             return ValueTask.CompletedTask;
         }
 
+        /// <summary>
+        /// Creates a new ICacheEntry, applicable only to IMemoryCache, and returns it for more advanced operations.
+        /// </summary>
+        /// <param name="key">The cache key this entry will utilize.</param>
+        public ValueTask<ICacheEntry> CacheAsync(object key) 
+            => ValueTask.FromResult(_cache.CreateEntry(key));
+
         /// <inheritdoc/>
-        public ValueTask<bool> TryGetAsync<TItem>(object key, out TItem value) 
-            => ValueTask.FromResult(_cache.TryGetValue(key, out value));
+        public ValueTask<TItem?> TryGetAsync<TItem>(object key)
+            => ValueTask.FromResult(_cache.TryGetValue(key, out TItem value) ? value : default);
 
         /// <inheritdoc/>
-        public ValueTask RemoveAsync(object key)
+        public ValueTask<TItem?> RemoveAsync<TItem>(object key)
         {
+            ValueTask<TItem?> value = TryGetAsync<TItem>(key);
+
             _cache.Remove(key);
 
-            return ValueTask.CompletedTask;
+            return value;
         }
+
+        // helper methods to fetch expirations from _options
+
+        private TimeSpan GetAbsoluteExpiration(Type type)
+            => _options.AbsoluteExpirations.ContainsKey(type)
+                ? _options.AbsoluteExpirations[type]
+                : _options.DefaultAbsoluteExpiration;
+
+        private TimeSpan GetSlidingExpiration(Type type) 
+            => _options.SlidingExpirations.ContainsKey(type)
+                ? _options.SlidingExpirations[type]
+                : _options.DefaultSlidingExpiration;
```

Comment:
I should've been more explicit in the notion to "make helper methods"; DRY would state that these should be put in the `CacheOptions` itself.

Reply:
implemented